polaris-core/src/main/java/org/apache/polaris/core/utils/CachedSupplier.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.utils;
+
+import java.util.function.Supplier;
+
+public class CachedSupplier<T> implements Supplier<T> {
```

Comment:
IIRC this functionality already exists in Guava.

Reply:
TIL - I believe you're talking about `Suppliers#memoize`. Thanks! Will convert to using this if we continue with this approach!